File: Entrada de Dados/Alfabetico/alfabetico.cs
using System;
using System.Linq;

class Program
{
  static void Main()
  {
    string TEXTO;
    bool VALIDACAO;

    do
    {
      Console.Write("Entre uma cadeia alfabetica: ");
      TEXTO = Console.ReadLine();
      VALIDACAO = TEXTO.Any(CARACTERE => (CARACTERE >= 'A' && CARACTERE <= 'Z') || 
                                         (CARACTERE >= 'a' && CARACTERE <= 'z') || 
                                          CARACTERE == ' ');
      if (!VALIDACAO)
      {
        Console.WriteLine("Entrada invalida. Por favor, tente novamente.");
      }
      else
      {
        break;
      }
    }
    while (true);

    Console.WriteLine("Voce informou a cadeia: " + TEXTO);
  }
}

File: Entrada de Dados/Alfabetico/alfabetico.pas
program ValidacaoCadeiaAlfabetica;

var
  TEXTO: string;
  I: integer;
  VALIDACAO: boolean;

begin
  repeat
    write('Entre uma cadeia alfabetica: ');
    readln(TEXTO);
    VALIDACAO := false;
    for I := 1 to length(TEXTO) do
    begin
      if (TEXTO[I] in ['A'..'Z', 'a'..'z', ' ']) then
      begin
        VALIDACAO := true;
      end
      else
      begin
        VALIDACAO := false;
        break;
      end;
    end;
    if (not VALIDACAO) then
      writeln('Entrada invalida. Por favor, tente novamente.')
    else
      break;
  until VALIDACAO;
  writeln('Voce informou a cadeia: ', TEXTO);
end.
